test/runtests.jl

```diff
@@ -74,11 +75,15 @@ end
 
 _props(cr::ConstRecord) = (:level => cr[:level], :msg => cr[:msg])
 
-Base.start(cr::ConstRecord) = start(_props(cr))
-Base.next(cr::ConstRecord, state) = next(_props(cr), state)
-Base.done(cr::ConstRecord, state) = done(_props(cr), state)
+if isdefined(Base, :iterate)
+    Base.iterate(cr::ConstRecord, state=1) = iterate(_props(cr), state)
```

Comment:
You can just do `state...` and pass it right through

Reply:
> and pass it right through

How do you mean?

Reply:
I see. Is that preferable?

Reply:
It doesn't assume iteration state, so yeah. Performance doesn't matter for this but I think there are splatting optimizations for small tuples.